twill-api/src/main/java/org/apache/twill/internal/TwillRuntimeSpecification.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.twill.internal;
+
+import org.apache.twill.api.TwillApplication;
+import org.apache.twill.api.TwillSpecification;
+
+/**
+ * Represents runtime specification of a {@link TwillApplication}.
+ */
+public class TwillRuntimeSpecification {
+
+  private final TwillSpecification twillSpecification;
+
+  private final String fsUser;
```

Comment:
Why everything are of `String` type? E.g. `twillRunId` should be a `RunId` type, `reservedMemory` should be `int`, `twillAppDir` should be an `URI`. Please use proper types.